### PR TITLE
Support multiple versions of Fast DDS in CI

### PR DESCRIPTION
**Pull request by rsanchez15**

### CODE REVIEW

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+jobs:
+
+  reusable_tests_v2:
+    name: reusable_tests_v2
+    uses: ./.github/workflows/reusable-workflow.yml
+    with:
+      custom_version_build: 'v2'
+      dependencies_artifact_postfix: '_nightly'
+
+  # Uncomment this block to run the tests for Fast DDS v3 when DDS Pipe is updated
```
Comment (irenebm) [inline .github/workflows/nightly.yml:17]:
```suggestion
  # Uncomment this block to run the tests for Fast DDS v3 when DDS Record & Replay is updated
```

File: .github/workflows/reusable-workflow.yml [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+#   - uncrustify
+#     - ubuntu-22.04
+#     - test uncrustify
+#
+#   - python-linter
+#     - ubuntu-22.04
+#     - test python linter
+#
+
+name: test
```
Comment (irenebm) [inline .github/workflows/reusable-workflow.yml:41]:
```suggestion
name: reusable-workflow
```

File: .github/workflows/reusable-workflow.yml [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+on:
+
+  workflow_call:
+    inputs:
+
+      custom_version_build:
+        description: >
+          Version of Fast DDS build from eProsima-CI.
+        required: true
+        type: string
```
Comment (irenebm) [inline .github/workflows/reusable-workflow.yml:52]:
```suggestion
        type: string
        default: 'v2'
```

File: .github/workflows/reusable-workflow.yml [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+      - name: Compile and run tests
+        uses: eProsima/eProsima-CI/multiplatform/colcon_build_test@v0
+        with:
+          packages_names: ${{ env.code_packages_names }}
+          cmake_args: -DBUILD_TESTS=ON -DCMAKE_BUILD_TYPE=${{ matrix.cmake_build_type }}
+          workspace_dependencies: './install'
+          ctest_args: --label-exclude "xfail"
+          colcon_meta_file: ./src/.github/workflows/configurations/${{ runner.os }}/colcon.meta
+
+
```
Comment (irenebm) [inline .github/workflows/reusable-workflow.yml:111]: add test report?

**Review by irenebm — CHANGES_REQUESTED** [review]

File: .github/workflows/reusable-workflow.yml (hunk 3 above)
Comment (rsanchez15) [reply .github/workflows/reusable-workflow.yml:52]: I prefer to keep it blank and make it required.

File: .github/workflows/reusable-workflow.yml (hunk 4 above)
Comment (rsanchez15) [reply .github/workflows/reusable-workflow.yml:111]: Maybe in another PR

**Review by irenebm — APPROVED** [review]
LGTM

**Pull request merged**